spark/src/main/scala/org/apache/comet/serde/arrays.scala

```diff
@@ -19,18 +19,23 @@
 
 package org.apache.comet.serde
 
-import org.apache.spark.sql.catalyst.expressions.{ArrayRemove, Expression}
+import org.apache.spark.sql.catalyst.expressions.{ArrayRemove, Attribute, Expression}
 import org.apache.spark.sql.types.{ArrayType, DataType, DataTypes, DecimalType, StructType}
 
 import org.apache.comet.CometSparkSessionExtensions.withInfo
+import org.apache.comet.serde.QueryPlanSerde.createBinaryExpr
 import org.apache.comet.shims.CometExprShim
 
-trait CometExpression {
-  def checkSupport(expr: Expression): Boolean
+trait CometExpressionSerde {
```

Comment:
This is presumably not just for arrays, so perhaps should be in a common file?

Reply:
Yes, currently, there is only one implementation, which is why it is here at the moment, but this can move to a common class. Perhaps I should just move this into QueryPlanSerde source file

Reply:
I moved the trait and added some scaladocs.